src/ProgressOnderwijsUtils/MetaObject/MetaObjectBulkCopy.cs

```diff
@@ -23,22 +22,48 @@ public static class MetaObjectBulkCopy
         /// <typeparam name="T">The type of metaobject to be inserted</typeparam>
         /// <param name="metaObjects">The list of entities to insert</param>
         /// <param name="sqlconn">The Sql connection to write to</param>
-        /// <param name="tableName">The name of the table to import into; must be a valid sql identifier (i.e. you must escape special characters if any).</param>
-        public static void BulkCopyToSqlServer<T>([NotNull] this IEnumerable<T> metaObjects, [NotNull] SqlCommandCreationContext sqlconn, [NotNull] string tableName) where T : IMetaObject, IPropertiesAreUsedImplicitly
+        /// <param name="table">The table, including schema information, to write the entities into.</param>
+        public static void BulkCopyToSqlServer<T>([NotNull] this IEnumerable<T> metaObjects, [NotNull] SqlCommandCreationContext sqlconn, [NotNull] DatabaseDescription.Table table)
+            where T : IMetaObject, IPropertiesAreUsedImplicitly
         {
+            metaObjects.BulkCopyToSqlServer(sqlconn, table.QualifiedName, table.Columns);
+        }
+
+
+        public static void BulkCopyToSqlServer<T>([NotNull] this IEnumerable<T> metaObjects, [NotNull] SqlCommandCreationContext sqlconn, [NotNull] string tableName, [NotNull] DatabaseDescription.TableColumn[] columns)

```

Comment:
Het is denk ik zinnig om een overload te hebben op basis van DatabaseDescription, en eentje op basis van tabel-naam+schema, maar deze zit er net ongelukkig tussenin; want DatabaseDescription.TableColumn kun je eigenlijk alleen hebben als je al een DatabaseDescription hebt.

Dus daarom zou ik ervoor pleiten hier `DbColumnMetaData` ipv `TableColumn` mee te geven.  Ik zou zelfs `ColumnDefinition` overwegen, maar daar moeten we dan weer opletten dat je evt. ooit nog dingen als IsComputed en IsIdentity oid nog erbij zou moeten weten te doen.

Reply:
In principe kan ik me niet een hele goede reden verzinnen om ooit zowel een tabel naam als ook `DatabaseDescription.TableColumn` types tegelijk te behandelen - of je hebt de full utility package (makkelijk mee te werken, maar kun je niet goed zelf aan maken), of je wilt iets makkelijk aan te maken (maar dus geen DatabaseDescription afhankelijkheid), maar niet allebei.

Reply:
Vaak ook niet eh, wij gebruiken temp tables ook om willekeurige objecten in de DB "even" te kunnen gebruiken - of die nou in een andere tabel passen of niet.

Reply:
Zie usages van `CopyMetaObjectsToNewTable` in progress; en ik gok dat we dat ook nog wel meer zouden willen gebruiken als het makkelijk&snel was.

Reply:
Ik maak er wel DbColumnMetaData's van.